tools/restatectl/src/lambda.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::HashMap;
+
+use lambda_runtime::LambdaEvent;
+
+#[derive(serde::Deserialize)]
+struct Request {
+    args: Vec<String>,
+    #[serde(default)]
+    env: Option<HashMap<String, String>>,
+}
+
+#[derive(serde::Serialize)]
+struct Response {
+    status: i32,
+    stdout: String,
+    stderr: String,
+}
+
+pub async fn handle_lambda_if_needed() {
+    if !std::env::var("AWS_LAMBDA_FUNCTION_NAME").is_ok() {
+        return;
+    }
+
+    let Ok(exe) = std::env::current_exe() else {
+        return;
+    };
+
+    let Some("bootstrap") = exe.file_name().and_then(std::ffi::OsStr::to_str) else {
+        return;
+    };
+
+    let func = lambda_runtime::service_fn(|event: LambdaEvent<Request>| async {
+        eprintln!("Executing restatectl {}", event.payload.args.join(","));
+
+        let mut cmd = tokio::process::Command::new(exe.as_os_str());
+
+        cmd.args(event.payload.args);
+
+        for (key, value) in event.payload.env.unwrap_or_default() {
+            cmd.env(key, value);
+        }
+
+        cmd.env_remove("AWS_LAMBDA_FUNCTION_NAME"); // so that we don't just hit this code path again
+
+        cmd.stdout(std::process::Stdio::piped());
+        cmd.stderr(std::process::Stdio::piped());
+
+        let child = cmd.spawn()?;
+        let output = child.wait_with_output().await?;
+
+        let status = output.status.code().unwrap_or(-1);
+        let stdout = String::from_utf8_lossy(&output.stdout).into();
+        let stderr = String::from_utf8_lossy(&output.stderr).into();
```

Comment:
In a future revision it might be useful think about how we might send logs - or perhaps even everything - to CloudWatch for observability / audit.

Reply:
i think the eprintln covers this? shows up in my cloudwatch anyway